### ISSUE TITLE
Pluggable credentials decryption for VCAP_SERVICES data on CF

### ISSUE THREAD

**Issue by johnmcteague** [spring-cloud/spring-cloud-connectors#138]
To enhance security we are modifying all our service brokers to encrypt key pieces of data in the credentials map return as part of VCAP_SERVICES.

I started looking at how I could have a pluggable interface in CloudFoundryServiceInfoCreator that by default would not decrypt credentials, but could be replaced by one that decrypts based on our strategy.

Based on my initial analysis, as all ServiceInfoCreators are not created as part of the Spring lifecycle, but instead are through Java's ServiceLoader, there is no easy injection point that doesnt involve changes in several places that at this stage I'm not keen to embark on.

Is this something we forsee implementing or do you have pointers on best I could proceed to provide a PR?

**Comment by scottfrederick** [comment]
@johnmcteague This question is pretty old, my apologies for letting it slip through the cracks. The use of ServiceLoader would make it difficult to design a pluggable mechanism for decoding credentials from VCAP_SERVICES. There is a possibility that we could move to a Spring bean loading mechanism in a future release, but nothing like that is planned at this time. The only option I see right now is to replace the default `CloudFoundryServiceInfoCreator` implementations for the services you care about with custom implementations. 

Have you come up with a solution since creating this issue?

**Comment by johnmcteague** [comment]
We have simply created a custom set of ServiceInfoCreators that replace all the standard ServiceInfoCreators (MySql etc) where we have amended the service to use encrypted passwords. The complication we still have which we work around in a less-than-ideal scenario, is when we use these off-platform where our decryption service is not available. A pluggable scheme would have allowed us to switch out the decryption scheme, for now we use a bunch of if/else's to try and detect when we are running in CF.
